src/console/controllers/UsersController.php

```diff
@@ -81,6 +106,66 @@ public function actionListAdmins(): int
         return ExitCode::OK;
     }
 
+    /**
+     * Creates a user.
+     *
+     * @return int
+     */
+    public function actionCreate(): int
+    {
+        // Validate the arguments
+        $attributesFromArgs = ArrayHelper::withoutValue([
+            'email' => $this->email,
+            'username' => $this->username,
+            'newPassword' => $this->password,
+            'admin' => $this->admin,
+        ], null);
+
+        $user = new User($attributesFromArgs);
+
+        if (!$user->validate(array_keys($attributesFromArgs))) {
+            $this->stderr('Invalid arguments:' . PHP_EOL . '    - ' . implode(PHP_EOL . '    - ', $user->getErrorSummary(true)) . PHP_EOL, Console::FG_RED);
+            return ExitCode::USAGE;
+        }
+
+        if (Craft::$app->getConfig()->getGeneral()->useEmailAsUsername) {
+            $user->username = $this->email ?: $this->prompt('Email:', [
+                'required' => true,
+                'validator' => $this->_createInputValidator($user, 'email'),
+            ]);
+        } else {
+            $user->email = $this->email ?: $this->prompt('Email:', [
+                'required' => true,
+                'validator' => $this->_createInputValidator($user, 'email'),
+            ]);
+            $user->username = $this->username ?: $this->prompt('Username:', [
+                'required' => true,
+                'validator' => $this->_createInputValidator($user, 'username'),
+            ]);
+        }
+
+        $user->admin = $this->admin ?? $this->confirm('Make this user an admin?', $this->admin);
+
+        if ($this->password) {
+            $user->newPassword = $this->password;
+        } elseif ($this->interactive) {
+            $user->newPassword = $this->passwordPrompt([
+                'validator' => $this->_createInputValidator($user, 'newPassword'),
+            ]);
+        }
```

Comment:
Checking `$this->interactive`, as `$this->passwordPrompt` doesn't check for `$this->interactive` like prompt/confirm do… but maybe it should, just to avoid awkwardness?

Reply:
Looks like `yii\console\Controller::select` isn't `interactive` aware either, so I guess we can just leave as-is.